src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const transformedStates = [];
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        stateCopy = {
+          ...stateCopy,
+          ...extraData,
+        };
+        break;
+      case 'removeProperties':
+        const { keysToRemove } = action;
```

Comment:
you  could simplify it
```suggestion
    const { type, extraData, keysToRemove  } = action;

    switch (type) {
      case 'addProperties':
      
        stateCopy = {
          ...stateCopy,
          ...extraData,
        };
        break;
      case 'removeProperties':
```

Reply:
Thank you))